src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -0,0 +1,327 @@
+        public async Task TestMissingOnDefalutBreak_04()
+        {
+            await TestMissingAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        [||]switch (i)
+        {
+            case var x:
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestMissingOnAllBreak()
+        {
+            await TestMissingAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        [||]switch (i)
+        {
+            case 1:
+                break;
+            case 2:
+                break;
+            case 3:
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestMissingOnAllThrow()
+        {
+            await TestMissingAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        [||]switch (i)
+        {
+            case 1:
+                throw null;
+            default:
+                throw null;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestSingleAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        int j;
+        [||]switch (i)
+        {
+            case 1:
+                j = 4;
+                break;
+            case 2:
+                j = 5;
+                break;
+            case 3:
+                j = 6;
+                break;
+        }
+        throw null;
+    }
+}",
+@"class Program
+{
+    void M(int i)
+    {
+        var j = i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6,
+            _ => throw null
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestAssignment_UseBeforeAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        int j = 123;
+        M(i);
+        [||]switch (i)
+        {
+            case 1:
+                j = 4;
+                break;
+            case 2:
+                j = 5;
+                break;
+            case 3:
+                j = 6;
+                break;
+        }
+        throw null;
+    }
+}",
+@"class Program
+{
+    void M(int i)
+    {
+        int j = 123;
+        M(i);
+        j = i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6,
+            _ => throw null
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestMultiSingleAssignment()
```

Comment:
>TestMultiSingleAssignment [](start = 26, length = 25)

Could you add a similar test where assignments are not in the same order in each section? I see this is handled (won't trigger fix), but good to capture.

Reply:
Another good test to add would be when the right side of second assignment reads the value from the symbol assigned in the first assignment. For example:

```
case 1:
                j = 4;
                k = j;
                break;
```

---
In reply to: [242756030](https://github.com/dotnet/roslyn/pull/31812#discussion_r242756030) [](ancestors = 242756030)

Reply:
Actually, I believe we should not offer this diagnostic/fix if there is a read reference to any of the assigned symbols in subsequent statements in the switch section. Otherwise, it can lead to silent change of semantics, which would be undesirable.

---
In reply to: [246967939](https://github.com/dotnet/roslyn/pull/31812#discussion_r246967939) [](ancestors = 246967939,242756030)

Reply:
>it can lead to silent change of semantics, which would be undesirable.

it won't, because the switch must have a `default:` case which assigns the same set of variables.

Reply:
I am not sure I understand, can you please add a test for the following and verify no fix is offered?

```csharp
   int j = 3, k;
   switch (i)
   {
       case 1:
                j = 4;
                k = j;
                break;
       default:
                j = 5;
                k = j;
                break;
    }
```